File: cloud_testing/cloud_api_poc/src/main/java/cloud_api_poc/ToolResultsCallable.java
package cloud_api_poc;

import static cloud_api_poc.Utils.fatalError;

import com.google.api.services.toolresults.model.Step;
import com.google.testing.model.ToolResultsStep;
import java.util.List;
import java.util.concurrent.Callable;

class ToolResultsCallable implements Callable {
  private List<ToolResultsValue> toolResultsValue;
  private ToolResultsStep toolResultsStep;

  ToolResultsCallable(List<ToolResultsValue> toolResultsValue, ToolResultsStep toolResultsStep) {
    this.toolResultsValue = toolResultsValue;
    this.toolResultsStep = toolResultsStep;
  }

  private void run() {
    try {
      Step step =
          GcToolResults.get()
              .projects()
              .histories()
              .executions()
              .steps()
              .get(
                  toolResultsStep.getProjectId(),
                  toolResultsStep.getHistoryId(),
                  toolResultsStep.getExecutionId(),
                  toolResultsStep.getStepId())
              .execute();

      toolResultsValue.add(new ToolResultsValue(step, toolResultsStep));
    } catch (Exception e) {
      fatalError(e);
    }
  }

  @Override
  public Object call() throws Exception {
    run();
    return null;
  }
}

File: cloud_testing/kotlin_poc/src/main/kotlin/ToolResultsCallable.kt
import com.google.testing.model.ToolResultsStep
import java.util.concurrent.Callable

import Utils.fatalError

internal class ToolResultsCallable(private val toolResultsValue: MutableList<ToolResultsValue>,
                                   private val toolResultsStep: ToolResultsStep) : Callable<Any> {

    private fun run() {
        try {
            val step = GcToolResults.get()!!
                    .projects()
                    .histories()
                    .executions()
                    .steps()
                    .get(
                            toolResultsStep.projectId,
                            toolResultsStep.historyId,
                            toolResultsStep.executionId,
                            toolResultsStep.stepId)
                    .execute()

            toolResultsValue.add(ToolResultsValue(step, toolResultsStep))
        } catch (e: Exception) {
            fatalError(e)
        }

    }

    @Throws(Exception::class)
    override fun call(): Any? {
        run()
        return null
    }
}
